osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -684,7 +673,8 @@ protected void ApplyResult(Action<JudgementResult> application)
                     $"{GetType().ReadableName()} applied an invalid hit result (was: {Result.Type}, expected: [{Result.Judgement.MinResult} ... {Result.Judgement.MaxResult}]).");
             }
 
-            Result.TimeOffset = Math.Min(MaximumJudgementOffset, Time.Current - HitObject.GetEndTime());
+            Result.TimeAbsolute = Time.Current;
+            Result.TimeOffset = Math.Min(MaximumJudgementOffset, Result.TimeAbsolute - HitObject.GetEndTime());
```

Comment:
See this sort of thing is what screws me up about having both times settable independently. `TimeOffset` is clamped to `MaximumJudgementOffset`, but `TimeAbsolute` is stored raw without so much as looking at that property?

So if `ApplyResult()` is called after the maximum judgement offset is up, you can potentially have a situation wherein `HitObject.GetEndTime() + TimeOffset != TimeAbsolute` (fundamentally and mathematically, putting FP shenanigans aside), because only one of the two gets the clamp and `TimeAbsolute` is allowed to be *any* value, even potentially in excess of `HitObject.GetEndTime() + MaximumJudgementOffset`?

I really *really* do not like this. I won't block the PR over it if it's just me, but really this to me is the stuff of nightmares.

Reply:
What is the unclamped `TimeAbsolute` useful for? In other words, the purpose of the change was to decouple the property from the hitobject's start time, but this would still be true if the clamping where added back.

Reply:
This is bad and probably regress some cases described in the original `MaximumJudgementOffset` introduction PR #10962.
With #22292, it has access to `MaximumJudgementOffset` here so:

```csharp
internal double TrueTime;

double TimeOffset => Min(TrueTime - HitObject.GetEndTime(), HitObject.MaximumJudgementOffset);
double TimeAbsolute => Min(TrueTime, HitObject.GetEndTime() + HitObject.MaximumJudgementOffset);
```

Looks okay? (not sure about naming of `TrueTime`)

Reply:
The proposal above with one source-of-truth internal property is sitting much better with me, but it will not be as simple as that - there are test usages that will need updating.

But, as @smoogipoo already asked - do we *really* need `TrueTime` for anything? Maybe we can just set the proper value for now, and apply the proposal above separately in another PR?

Reply:
Here, `TrueTime` is needed to handle revert logic by a stack (last-in-first-out).
Alternatively, the set of judged results can be maintained by a sorted set and that will be unintuitive that: A judged then B judged, but A-result is reverted first then B-result reverted.

Reply:
I see.

I'm ok with `TrueTime` (maybe `RawTime` is better? likely needs xmldoc either way), but applying that won't be straightforward. There are test usages that set `TimeOffset` directly which will need to be migrated over.

Reply:
What's the go here? Are we going to rename the parameter still? I'd agree with `RawTime` more than `TrueTime`.

Reply:
Uh, I'm okay with adding `TrueTime` or `RawTime` as proposed, it just needs to be actually _applied_ here. And I don't see any changes for it in this PR, and as I said, it is not going to be as simple as the three lines above.

Reply:
Safe to say that that means things here are probably a tad too complex and I'm missing what is actually being proposed.

Feels like we need to take a step back and reconsider.